spec/controllers/holds_controller_spec.rb

```diff
@@ -350,14 +354,38 @@
         get :result, params: {}, session: { place_hold_catkey: '1', place_hold_results: results }
 
         placed_hold = assigns(:place_hold_results)[:success].first[:placed_hold]
-        expect(placed_hold.record.first['key']).to eq 'a_hold_key'
+        expect(placed_hold.record['key']).to eq 'a_hold_key'
       end
 
       it 'assigns failed holds results correctly' do
         get :result, params: {}, session: { place_hold_catkey: '1', place_hold_results: results }
 
         failed_hold = assigns(:place_hold_results)[:error].first[:failed_hold]
-        expect(failed_hold.record.first['fields']['barcode']).to eq 'not_holdable_barcode'
+        expect(failed_hold.record['fields']['barcode']).to eq 'not_holdable_barcode'
+      end
+
+      context 'when hold lookup does not return item level info' do
+        let(:hold_info_empty_item_response) { HOLD_LOOKUP_NIL_ITEM_RAW_JSON.to_json }
+        let(:parsed_hold_info) { JSON.parse hold_info_response }
+        let(:parsed_hold_info_empty_item) { JSON.parse hold_info_empty_item_response }
+        let(:hold_info_params) { ['a_hold_key', user[:session_token]] }
+        let(:mock_client) { instance_double(SymphonyClient) }
+
+        let(:results) { {
+          success: [{ barcode: 'holdable_barcode', hold_key: 'a_hold_key' }]
+        }.with_indifferent_access }
+
+        before do
+          allow(SymphonyClientParser).to receive(:parsed_response).with(mock_client, :get_hold_info, *hold_info_params)
+            .and_return(parsed_hold_info_empty_item, parsed_hold_info)
```

Comment:
So this is saying that the `SymphonyClientParser` would return two objects when `get_hold_info` is invoked? Am I reading that right? Is that possible? Wouldn't it always return just one hold record?

Reply:
No, it will return one object for each call, but first call will return the first arg, and the second will return the second one.

Reply:
https://relishapp.com/rspec/rspec-mocks/docs/configuring-responses/returning-a-value#specify-different-return-values-for-multiple-calls

Reply:
so SymphonyClientParser.parsed_response is called twice?

Reply:
wow, :the_more_you_know:

I don't think I've seen `and_return` used like this before, nice!